master/internal/api_workspace.go

```diff
@@ -179,17 +182,16 @@ func validateWorkspaceName(name string) error {
 // identifiable with regard to the corresponding workspace name and namespace within which the
 // determined installation was created while simultaneously guaranteeing uniqueness within a given
 // Kubernetes cluster, we use the following convention when auto-generating a namespace name:
-// <release.namespace>(19 chars) + "_" + <cluster_id>(4) + "_" + <workspace_name>(31) + "_" +
+// <workspace_name>(31) + "-" + <release.namespace>(19 chars) + "-" + <cluster_id>(4) + "-" +
 // <workspace_id>(6).
 // Note that we allow the workspace ID to occupy up to 6 characters (<= 999,999 unique
 // workspaces).
-func generateNamespaceName(clusterID string, workspace string, wkspID int) string {
+func generateNamespaceName(detInstallationNamespace, clusterID, workspaceName string, wkspID uint) string {
```

Comment:
nit (non-blocking): Taking a second look at this, can we pass in the function args in their "order of priority"? So 
```suggestion
func generateNamespaceName( workspaceName string, detInstallationNamespace, clusterID, wkspID uint) string {
```